From 91d8909add61215733e5d9599450cccc4d071274 Mon Sep 17 00:00:00 2001
From: Alessandro Decina <alessandro.decina@collabora.co.uk>
Date: Thu, 4 Aug 2011 09:49:46 +0200
Subject: [PATCH 037/119] mpegvideoparse: fix sequence header parsing

After bitrate there's 11 bits (1 marker bit + VBV buffer size) before the load
intra quantiser flag.
---
 gst/videoparsers/mpegvideoparse.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/gst/videoparsers/mpegvideoparse.c b/gst/videoparsers/mpegvideoparse.c
index 45f8dd3..e85d77b 100644
--- a/gst/videoparsers/mpegvideoparse.c
+++ b/gst/videoparsers/mpegvideoparse.c
@@ -209,6 +209,10 @@ gst_mpeg_video_params_parse_sequence (MPEGVParams * params, GstBitReader * br)
     params->bitrate *= 400;
   }
 
+  /* skip 1 + VBV buffer size */
+  if (!gst_bit_reader_skip (br, 11))
+    goto failed;
+
   /* constrained_parameters_flag */
   GET_BITS (br, 1, &bits);
 
-- 
1.7.5.4

